azure-proto-core-test/SystemAssignedIdentityTests.cs

```diff
@@ -0,0 +1,348 @@
+                json = f.ReadToEnd();
+            JsonDocument document = JsonDocument.Parse(json);
+            JsonElement rootElement = document.RootElement;
+            var identityJsonProperty = rootElement.EnumerateObject().First();
+            Assert.Throws<FormatException>(delegate { Deserialize(identityJsonProperty.Value); });
+        }
+
+        [TestCase]
+        public void TestDeserializerOneEmptyString()
+        {
+            string json = "";
+            string path = ((AssemblyMetadataAttribute)GetType().Assembly.GetCustomAttribute(typeof(AssemblyMetadataAttribute))).Value;
+            using (StreamReader f = new StreamReader(path + "/TestAssets/SystemAssignedIdentity/SystemAssignedOneEmptyString.json"))
+                json = f.ReadToEnd();
+            JsonDocument document = JsonDocument.Parse(json);
+            JsonElement rootElement = document.RootElement;
+            var identityJsonProperty = rootElement.EnumerateObject().First();
+            Assert.Throws<FormatException>(delegate { Deserialize(identityJsonProperty.Value); });
+        }
+
+        [TestCase]
+        public void TestDeserializerTenantIdValueNull()
+        {
+            string json = "";
+            string path = ((AssemblyMetadataAttribute)GetType().Assembly.GetCustomAttribute(typeof(AssemblyMetadataAttribute))).Value;
+            using (StreamReader f = new StreamReader(path + "/TestAssets/SystemAssignedIdentity/SystemAssignedOneValueNull.json"))
+                json = f.ReadToEnd();
+            JsonDocument document = JsonDocument.Parse(json);
+            JsonElement rootElement = document.RootElement;
+            var identityJsonProperty = rootElement.EnumerateObject().First();
+            Assert.Throws<InvalidOperationException>(delegate { Deserialize(identityJsonProperty.Value); });
+        }
+
+        [TestCase]
+        public void TestDeserializerPrincipalIdValueNull()
+        {
+            string json = "";
+            string path = ((AssemblyMetadataAttribute)GetType().Assembly.GetCustomAttribute(typeof(AssemblyMetadataAttribute))).Value;
+            using (StreamReader f = new StreamReader(path + "/TestAssets/SystemAssignedIdentity/SystemAssignedOneOtherValueNull.json"))
+                json = f.ReadToEnd();
+            JsonDocument document = JsonDocument.Parse(json);
+            JsonElement rootElement = document.RootElement;
+            var identityJsonProperty = rootElement.EnumerateObject().First();
+            Assert.Throws<InvalidOperationException>(delegate { Deserialize(identityJsonProperty.Value); });
+        }
+
+        [TestCase]
+        public void TestDeserializerTenantIdInvalid()
+        {
+            string json = "";
+            string path = ((AssemblyMetadataAttribute)GetType().Assembly.GetCustomAttribute(typeof(AssemblyMetadataAttribute))).Value;
+            using (StreamReader f = new StreamReader(path + "/TestAssets/SystemAssignedIdentity/SystemAssignedInvalid.json"))
```

Comment:
pretty sure you can use "./TestAssets/SystemAssignedIdentity/SystemAssignedInvalid.json" since the file will be copied local along side the binaries.

if this isn't happening then you need to set the content property of the json file

Also there is a simplified version of this
var json = File.ReadAllText("./TestAssets/SystemAssignedIdentity/SystemAssignedInvalid.json");